sycl/test-e2e/LLVMIntrinsicLowering/bitreverse.cpp

```diff
@@ -0,0 +1,189 @@
+// Test that llvm.bitreverse is lowered correctly by llvm-spirv
+
+// RUN: %{build} -o %t.O2.out -O2
+// RUN: %{run} %t.O2.out | FileCheck %s
+
+// CHECK: Passed bitreverse  8  1
```

Comment:
Instead of checking for a string, let's use the exit code. If the test fails, we should return non-zero from main, and that will cause the test to fail. Feel free to have code that prints out some info when the result is wrong though. Those should be able to use normal host printfs instead of the device one.

Reply:
Updated to use exit code.